### PR TITLE
Create a new test suite in CI pipeline for admin GUI tests 

### LINKED ISSUE #21885: Create a new test suite in CI pipeline for admin GUI tests
The task is to create a script and make necessary changes in the devtests to make the new test suite run on hudson.

### PR DESCRIPTION
**Pull request by kokil12**
Fixes #21885 : Adding a script run_test.sh and doing necessary changes to make the admingui devtests run on hudson.

### CODE REVIEW

**Comment by sumauppa** [comment]
@glassfishrobot Run CI tests please

**Comment by kokil12** [comment]
@glassfishrobot Run CI tests please

File: appserver/admingui/devtests/run_test.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  JUD=$WORKSPACE/results/junitreports/test_results_junit.xml
+  rm -f ${JUD} || true
+  echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" >> ${JUD}
+  echo "<testsuites>" >> ${JUD}
+  find ${JUD_DIR} -name "*.xml" -type f -exec cat '{}' \; | ${SED} 's/<?xml version=\"1.0\" encoding=\"UTF-8\" ?>//g' >> ${JUD}
+  echo -e "\n</testsuites>" >> ${JUD}
+}
+
+test_run(){
+  touch /tmp/password.txt
```
Comment (yaminikb) [inline appserver/admingui/devtests/run_test.sh]: I see all tests generally create the password file in APS_HOME, lets be consistent.

File: appserver/admingui/devtests/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  chmod 600 /tmp/password.txt
+  echo "AS_ADMIN_PASSWORD=" > /tmp/password.txt
+  echo "AS_ADMIN_NEWPASSWORD=admin" >> /tmp/password.txt
```
Comment (yaminikb) [inline appserver/admingui/devtests/run_test.sh]: We shouldn't hard code passwords, we should generate a random one and use that.

File: appserver/admingui/devtests/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+run_test_id(){
+  #a common util script located at main/appserver/tests/common_test.sh
+  source `dirname $0`/../../tests/common_test.sh
+  kill_process
+  delete_gf
+  export M2_HOME=/net/gf-hudson/scratch/gf-hudson/export2/hudson/tools/apache-maven-3.0.3
+  export MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=384m"
+  export MAVEN_REPO=$WORKSPACE/repository
+  export MAVEN_SETTINGS=$M2_HOME/settings-nexus.xml
+  export PATH=$M2_HOME/bin:$JAVA_HOME/bin:$PATH
```
Comment (yaminikb) [inline appserver/admingui/devtests/run_test.sh]: 1. M2_HOME, PATH is already set in common_test.sh, no need to set here.
2. Did you mean to set M2_REPO above? MAVEN_REPO is not used.

**Review by yaminikb — CHANGES_REQUESTED** [review]

File: appserver/admingui/devtests/run_test.sh (hunk 2 above)
Comment (sumauppa) [inline appserver/admingui/devtests/run_test.sh]: In GUI dev tests if secure admin is on, user name and passwords are hardcoded. We need these values if secure admin is on. That is the reason Kokil added these lines.

File: appserver/admingui/devtests/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+#
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+#
+# The contents of this file are subject to the terms of either the GNU
+# General Public License Version 2 only ("GPL") or the Common Development
+# and Distribution License("CDDL") (collectively, the "License").  You
+# may not use this file except in compliance with the License.  You can
+# obtain a copy of the License at
+# https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
```
Comment (yaminikb) [inline appserver/admingui/devtests/run_test.sh]: Fix the copyright text

**Review by yaminikb — CHANGES_REQUESTED** [review]

**Comment by kokil12** [comment]
Fixed the copyright text.

**Comment by kokil12** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]

**Pull request merged**